### DISCUSSION TITLE
Consider making contextvars required on Python 3.6

### DISCUSSION THREAD

**Discussion by JayH5** [encode/starlette#1269] [Ideas] [answered]
We're having to do some contortions (see `concurrency` module or #1257, #1258, #1260) to allow `contextvars` to not be installed. Rather, let's just make it a requirement on Python 3.6, because:
* It's only required on Python 3.6, which is EoL at the end of 2021. I don't think it's worth putting much effort into reducing one small dependency for this version.
* Lots of things (Trio, various tracing libraries) depend on `contextvars` already.

**Comment by graingert** [comment, marked as answer]
it's already there https://github.com/encode/starlette/blob/master/setup.py#L43

**Reply by JayH5** [reply]
Ah I'm not very smart and mixed up my `context<thing>`s. I meant `contextvars`, not `contextlib2`.
